rskj-core/src/main/java/org/ethereum/vm/program/invoke/ProgramInvokeImpl.java

```diff
@@ -98,16 +100,19 @@ public ProgramInvokeImpl(DataWord address, DataWord origin, DataWord caller, Dat
         this.byTransaction = false;
         this.callDeep = callDeep;
         this.blockStore = blockStore;
+        this.isStaticCall = isStaticCall;
         this.byTestingSuite = byTestingSuite;
     }
 
     public ProgramInvokeImpl(byte[] address, byte[] origin, byte[] caller, byte[] balance,
                              byte[] gasPrice, byte[] gas, byte[] callValue, byte[] msgData,
                              byte[] lastHash, byte[] coinbase, long timestamp, long number, int transactionIndex, byte[] difficulty,
                              byte[] gaslimit,
-                             Repository repository, BlockStore blockStore, boolean byTestingSuite) {
+                             Repository repository, BlockStore blockStore,
```

Comment:
Why don't you set the `isStaticCall` flag in these two constructors? The best pattern here would be to chain constructor calls like this:

```java
public ProgramInvokeImpl(byte[] address, byte[] origin, byte[] caller, byte[] balance,
                         byte[] gasPrice, byte[] gas, byte[] callValue, byte[] msgData,
                         byte[] lastHash, byte[] coinbase, long timestamp, long number, int transactionIndex, byte[] difficulty,
                         byte[] gaslimit,
                         Repository repository, BlockStore blockStore) {

    this(
            new DataWord(address),
            new DataWord(origin),
            new DataWord(caller),
            new DataWord(balance),
            new DataWord(gasPrice),
            Program.limitToMaxLong(new DataWord(gas)),
            new DataWord(callValue),
            msgData,
            // last Block env
            new DataWord(lastHash),
            new DataWord(coinbase),
            new DataWord(timestamp),
            new DataWord(number),
            new DataWord(transactionIndex),
            new DataWord(difficulty),
            new DataWord(gaslimit),
            repository,
            0,
            blockStore,
            false
    );
}
```

Then, changing the base constructor will always force you to change the other ones.

EDIT: I saw this is the way EthereumJ does things. We can leave it as it is, but I think we can do better :muscle:

Reply:
Why add a call to the stack if the other constructor will get deprecated and never used ever again?

Reply:
Add another call to the stack doesn't really matter in this context. You should favor other qualities before prematurely optimizing: readability, maintainability, correctness.

On the other hand, if the other constructor is deprecated we should delete it.